src/sentry/snuba/query_subscriptions/run.py

```diff
@@ -0,0 +1,143 @@
+            from sentry.utils import metrics
+
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=random() <= options.get("subscriptions-query.sample-rate"),
+            ), metrics.timer(
+                "snuba_query_subscriber.handle_message", tags={"dataset": self.dataset.value}
+            ):
+                value = message.value
+                assert isinstance(value, BrokerValue)
+                offset = value.offset
+                partition = value.partition.index
+                message_value = value.payload.value
+                try:
+                    handle_message(
+                        message_value,
+                        offset,
+                        partition,
+                        self.topic,
+                        self.dataset.value,
+                        self.jsoncodec,
+                    )
+                except Exception:
+                    # This is a failsafe to make sure that no individual message will block this
+                    # consumer. If we see errors occurring here they need to be investigated to
+                    # make sure that we're not dropping legitimate messages.
+                    logger.exception(
+                        "Unexpected error while handling message in QuerySubscriptionStrategy. Skipping message.",
+                        extra={
+                            "offset": offset,
+                            "partition": partition,
+                            "value": message_value,
+                        },
+                    )
+
+        return RunTaskWithMultiprocessing(
+            process_message,
+            CommitOffsets(commit),
+            self.num_processes,
+            self.max_batch_size,
+            self.max_batch_time,
+            self.input_block_size,
+            self.output_block_size,
+            initializer=initialize_consumer_state,
+        )
+
+
+def get_query_subscription_consumer(
+    topic: str,
+    group_id: str,
+    strict_offset_reset: bool,
+    initial_offset_reset: str,
+    max_batch_size: int,
+    max_batch_time: int,
+    processes: int,
+    input_block_size: int,
+    output_block_size: int,
+) -> StreamProcessor[KafkaPayload]:
+    from django.conf import settings
+
+    from sentry.utils import kafka_config, metrics
+
+    cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+    cluster_options = kafka_config.get_kafka_consumer_cluster_options(cluster_name)
+    consumer = KafkaConsumer(
+        build_kafka_consumer_configuration(
+            cluster_options,
+            group_id=group_id,
+            strict_offset_reset=strict_offset_reset,
+            auto_offset_reset=initial_offset_reset,
+        )
+    )
+    metrics_wrapper = MetricsWrapper(metrics.backend, name="query_subscription_consumer")
+    configure_metrics(metrics_wrapper)
```

Comment:
you will need to do this in the subprocess as well. Take a look at https://github.com/getsentry/sentry/blob/f8c0324a0b0d49e110401468cc665d6c4cda0904/src/sentry/sentry_metrics/configuration.py#L169

we should (long-term) change arroyo to not make this necessary though, IMO